### PR TITLE
falcon: 2013-09-19 -> unstable-2018-10-23

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 108040` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>falcon</li>
  </ul>
</details>

**Review by prusnak — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108040` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>falcon</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

falcon.log:
<details>
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/development/interpreters/falcon/default.nix:19:5:

```
   |
19 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>

**Comment by davidak** [comment]
@siraben can you clarify the license as the comment from @SuperSandro2000 tells?

**Comment by prusnak** [comment]
According to https://github.com/falconpl/falcon/blob/637e2d5cd950a874496042993c02ab7d17c1b688/LICENSE#L286-L289  the license should be `licenses. gpl2Only`

**Comment by prusnak** [comment]
@GrahamcOfBorg build falcon

**Comment by prusnak** [comment]
The build fails on x86_64-linux - https://logs.nix.ci/?key=nixos/nixpkgs.108040&attempt_id=878676a0-3d64-47d4-98a7-534a4839741c

**Comment by siraben** [comment]
Interesting, the build succeeded for both me and @SuperSandro2000 on x86_64-linux on 2020-12-31.

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 108040` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>falcon</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> What should I adjust the hardening flags to?

Probably the issue is similar to the packages listed here https://search.nix.gsc.io/?q=narrowing&i=nope&files=&repos=

**Comment by siraben** [comment]
@SuperSandro2000 @prusnak @davidak please take another look

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 108040` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>falcon</li>
  </ul>
</details>

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 108040` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>falcon</li>
  </ul>
</details>

File: pkgs/development/interpreters/falcon/default.nix [hunk 1]
```diff
@@ -1,14 +1,14 @@
-{ lib, stdenv, fetchFromGitHub, cmake, pkg-config, pcre, zlib, sqlite }:
+{ lib, gccStdenv, fetchFromGitHub, cmake, pkg-config, pcre, zlib, sqlite }:
```
Comment (davidak) [inline pkgs/development/interpreters/falcon/default.nix:1]: @siraben why do you use `gccStdenv` instead of `stdenv`?

**Review by davidak — APPROVED** [review]
Change looks OK

Builds, seem to work

```
[nix-shell:~/.cache/nixpkgs-review/pr-108040-1]$ falcon -v
Falcon compiler and interpreter.
Version 0.9.6.9 (Chimera)
```

File: pkgs/development/interpreters/falcon/default.nix (hunk 1 above)
Comment (siraben) [reply pkgs/development/interpreters/falcon/default.nix:1]: Otherwise the build fails on darwin with clang.

**Pull request merged**